### DISCUSSION TITLE
The form of the basis function

### DISCUSSION THREAD

**Discussion by nuomi68** [idaholab/moose#20624] [Q&A General] [answered]
How do I change the form of the basis function ? What is its default value? I didn't find  instructions on the official website.I hope someone can answer that,thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can change it for every variable in the `[Variables]` block. If you want a new one we currently dont support, you ll have to develop in libmesh
see
https://mooseframework.inl.gov/source/variables/MooseVariable.html

Please note that not every basis works for every problem, some kernels wont expect some exotic basis and you'll probably get a seg fault on the most exotic ones. 

Guillaume

**Reply by nuomi68** [reply]
It's been very helpful. Thank you very much
